### ISSUE TITLE
Webdav not working

### ISSUE THREAD

**Issue by FeralFlora** [innomatica/carta#1]
Hi, thanks for this

**Comment by innomatica** [comment]
WebDAV servers tend to have different internal directory structures. Current version is tested with Apache modDAV, rclone server, and NextCloud. And they all have different directory conventions. 

In case of Koofr, it looks like you have to enter

`https://app.koofr.net/`

for your server URL then 

`dav/Koofr`

for your directory according to their [document](https://app.koofr.net/help/webdav#:~:text=Type%20https%3A%2F%2Fapp.koofr,using%20different%20credentials%20check%20box).

**Comment by FeralFlora** [comment]
> for your directory according to their [document](https://app.koofr.net/help/webdav#:~:text=Type%20https%3A%2F%2Fapp.koofr,using%20different%20credentials%20check%20box).

Yes, I did follow the Koofr docs. I've set it up quite a few times before, so I am familiar with the config. However, in the docs, everything including `dav/Koofr` is provided as the webdav URL, and then you can add the specific sub-directory as the directory to access. That's what I tried first (see screenshots below).

## My first approach

So I used `site url: https://app.koofr.net/dav/Koofr `
and `directory: /My media/Audiobooks`.

![Screenshot_20240522-111047](https://github.com/innomatica/carta/assets/33608083/49d61e40-fe42-43fd-b902-3371388237d5)

And as I mentioned, when I do it like this, I can see all the audiobook folders under `My media/Audiobooks`. 

![Screenshot_20240522-111029](https://github.com/innomatica/carta/assets/33608083/bbfe4538-fb63-470b-829c-2d937ab5a293)

This issue occurs when I try to select a folder.

![Screenshot_20240522-111115](https://github.com/innomatica/carta/assets/33608083/3b375c8a-55b2-4282-841f-fd735a3b5252)

## Trying your approach

![Screenshot_20240522-142203](https://github.com/innomatica/carta/assets/33608083/69e40731-5ea5-4c6d-a13c-b34f6790c240)

When I try your suggested approach of using `dav/Koofr` as the directory, it does not work. As soon as I press "Koofr" in the "add" menu, I get the "Failed to connect to the server" error.

![Screenshot_20240522-142212](https://github.com/innomatica/carta/assets/33608083/f7231359-4d70-4ac4-9338-ad33af1ef5d6)

## Another approach

Lastly, I tried a third setup option, which is:

`site url: https://app.koofr.net/dav`
`directory: /Koofr`

With this setup, the connection works, and the folders of the Koofr root are displayed:
![Screenshot_20240522-111135](https://github.com/innomatica/carta/assets/33608083/7ea740e2-6746-408e-a6ad-d93afcc329c1)

However, if I select any folder, I get them same error as before:
![Screenshot_20240522-142337](https://github.com/innomatica/carta/assets/33608083/8bee8a6d-a28c-4d59-95f2-522906c9b51f)

---

In conclusion, it seems to me that this is not a webdav configuration error, because even in the two scenarios where the initial connection works and the directory is displayed, no interaction with the folders is possible, but rather results in errors.

**Comment by innomatica** [comment]
Interesting. Let me dig into this.

**Comment by innomatica** [comment]
Please test version 2.6.1+46. And Let me know if it works for you.

**Comment by FeralFlora** [comment]
> Please test version 2.6.1+46. And Let me know if it works for you.

It works now! The only thing I am missing now is to retain the ability to add a directory, so I don't have to navigate from the drive root every time I want to add a book.

Oh yeah, and everytime I open the Koofr webdav settings to tweak them, the server type defaults to nextcloud, so I have to reselect Koofr.

**Comment by FeralFlora** [comment]
> Please test version 2.6.1+46. And Let me know if it works for you.

It works to select books now! However, I miss the ability to select a specific directory, so you don't have to navigate from the root each time.

Oh yeah, and the server type defaults to nextcloud each time I open the Koofr settings, so I have reselect Koofr each timec

**Comment by innomatica** [comment]
Version 2.6.2+47 should address above issues.
